diff --git a/admin_toolbar/admin_toolbar_tools/src/Controller/ToolbarController.php b/admin_toolbar/admin_toolbar_tools/src/Controller/ToolbarController.php
index 478eeb0..2a1d530 100644
--- a/admin_toolbar/admin_toolbar_tools/src/Controller/ToolbarController.php
+++ b/admin_toolbar/admin_toolbar_tools/src/Controller/ToolbarController.php
@@ -288,7 +288,7 @@ class ToolbarController extends ControllerBase {
    */
   public function themeRebuild() {
     $this->themeRegistry->reset();
-    $this->messenger()->addMessage($this->t('Theme registry rebuilded.'));
+    $this->messenger()->addMessage($this->t('Theme registry rebuilt'));
     return new RedirectResponse($this->reloadPage());
   }
 
